src/core/geometry/geom3/intersection.js

```diff
@@ -42,10 +33,20 @@ const intersectSub = (ohterCsg, geometry, doRetesselate, doCanonicalize) => {
   a.addPolygons(b.allPolygons())
   a.invert()
   let result = fromPolygons(a.allPolygons())
-  if (doRetesselate) result = retesselate(result)
   if (doCanonicalize) result = canonicalize(result)
+  if (doRetesselate) result = retessellate(result)
+  return result
+}
+
+const intersect = (otherGeom3, ...geometries) => {
+  let result = otherGeom3
```

Comment:
same remark as for subtract, you need to ensure that geometries is an array !
as it is `...geometries` syntax unpacks an array of geometries (please take a look at rest/spread syntax for es6)

Reply:
Are you sure?

Perhaps I've misunderstood, but I don't think that ...geometries will unpack a passed array, but rather will pack a local array.

e.g., ((...x) => x)([1, 2]) produces [ [ 1, 2 ] ] for me.

The [1, 2] isn't unpacked, but is rather packed into an array as [[1, 2]]